### ISSUE TITLE
Error: module is not defined

### ISSUE THREAD

**Issue by ben-ekw** [fidian/fid-umd#6]
Cool project! However, when I try to use it via the CLI, the output is different than your README. Steps to replicate:

1.  `npm install -g fid-umd`
2. `echo // *** Your JavaScript is here *** > example.js`
3. `fid-umd example.js`

Result:

```
// fid-umd {"name":"Unknown"}
(function (name, root, factory) {
    function isObject(x) { return typeof x === "object"; }
    if (isObject(module) && isObject(module.exports)) {
        module.exports = factory();
    } else if (isObject(exports)) {
        exports[name] = factory();
    } else if (isObject(root.define) && root.define.amd) {
        root.define(name, [], factory);
    } else if (isObject(root.modulejs)) {
        root.modulejs.define(name, factory);
    } else if (isObject(root.YUI)) {
        root.YUI.add(name, function (Y) { Y[name] = factory(); });
    } else {
        root[name] = factory();
    }
}("Unknown", this, function () {
    // fid-umd end
    // *** Your JavaScript is here ***
    // fid-umd post
}));
// fid-umd post-end
```

Your README indicates it should have been:

```
// fid-umd {"name":"Unknown"}
(function (name, root, factory) {
    function isObject(x) { return typeof x === "object"; }
    if (isObject(root.module) && isObject(root.module.exports)) {
        root.module.exports = factory();
    } else if (isObject(root.exports)) {
        root.exports[name] = factory();
    } else if (isObject(root.define) && root.define.amd) {
        root.define(name, [], factory);
    } else if (isObject(root.modulejs)) {
        root.modulejs.define(name, factory);
    } else if (isObject(root.YUI)) {
        root.YUI.add(name, function (Y) { Y[name] = factory(); });
    } else {
        root[name] = factory();
    }
}("Unknown", this, function () {
    // fid-umd end
    // *** Your JavaScript is here ***
    // fid-umd post
}));
// fid-umd post-end
```

Diff: https://www.diffchecker.com/rguNfuu6/

When I try to use it in an AMD project it throws this error: `ReferenceError: module is not defined`

**Comment by fidian** [comment]
Looks like there are two problems.

First, I neglected to update the README when I made a change in 2015. Node stopped allowing `this.module` to reference the built-in `module` variable.

Secondly, I don't have a good way to test for current Node.js environments without using `module`, which breaks AMD according to your report. Is there a minimal AMD project that I can clone in order to test various workarounds? Without one, I might be stuck on this issue.
